oci/oci.go

```diff
@@ -148,6 +148,56 @@ func (r *Runtime) StartContainer(c *Container) error {
 	return nil
 }
 
+// ExecSync execs a command in a container and returns it's stdout, stderr and return code.
+func (r *Runtime) ExecSync(c *Container, command []string, timeout int64) (stdout []byte, stderr []byte, exitCode int32, err error) {
+	args := []string{"exec", c.name}
+	args = append(args, command...)
+	cmd := exec.Command(r.Path(), args...)
+	var stdoutBuf, stderrBuf bytes.Buffer
+	cmd.Stdout = &stdoutBuf
+	cmd.Stderr = &stderrBuf
+	err = cmd.Start()
+	if err != nil {
+		return stdoutBuf.Bytes(), stderrBuf.Bytes(), -1, err
+	}
+
+	if timeout > 0 {
+		done := make(chan error, 1)
+		go func() {
+			done <- cmd.Wait()
+		}()
+
+		select {
+		case <-time.After(time.Duration(timeout) * time.Second):
+			err = unix.Kill(cmd.Process.Pid, syscall.SIGKILL)
+			if err != nil && err != syscall.ESRCH {
+				return stdoutBuf.Bytes(), stderrBuf.Bytes(), -1, fmt.Errorf("failed to kill process on timeout: %v", err)
+			}
+			return stdoutBuf.Bytes(), stderrBuf.Bytes(), -1, fmt.Errorf("command timed out")
+		case err = <-done:
+			if err != nil {
+				if exitErr, ok := err.(*exec.ExitError); ok {
+					if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+						return stdoutBuf.Bytes(), stderrBuf.Bytes(), int32(status.ExitStatus()), err
+					}
+				}
```

Comment:
should we return the error even if it's not an `ExitError` for debugging and understanding what happened?